src/hotspot/os/posix/os_posix.cpp

```diff
@@ -716,9 +716,20 @@ void* os::dll_lookup(void* handle, const char* name) {
 }
 
 void os::dll_unload(void *lib) {
-  const char* l_path = LINUX_ONLY(os::Linux::dll_path(lib))
-                       NOT_LINUX("<not available>");
-  if (l_path == NULL) l_path = "<not available>";
+  // os::Linux::dll_path returns a pointer to a string that is owned by the dynamic loader. Upon
+  // calling dlclose the dynamic loader may free the memory containing the string, thus we need to
+  // copy the string to be able to reference it after dlclose.
+  const char* l_path = NULL;
+  char* l_pathdup = NULL;
+  LINUX_ONLY(
```

Comment:
The XXX_ONLY macros are intended for one-liners or in-line conditionals. For a block please use `#ifdef LINUX`.

Reply:
Ah, thanks for the correction. Was not sure. Fixed.